sdk/eventhub/azure-eventhub/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,228 @@
+
+### "...current receiver '<RECEIVER_NAME>' with epoch '0' is getting disconnected"
+
+The entire error message looks something like this:
+
+> New receiver 'nil' with higher epoch of '0' is created hence current receiver 'nil' with epoch '0'
+> is getting disconnected. If you are recreating the receiver, make sure a higher epoch is used.
+> TrackingId:<GUID>, SystemTracker:<NAMESPACE>:eventhub:<EVENT_HUB_NAME>|<CONSUMER_GROUP>,
+> Timestamp:2022-01-01T12:00:00}"}
+
+This error is expected when load balancing occurs after EventProcessorClient instances are added or removed.  Load balancing is an ongoing process.  When using the BlobCheckpointStore with your consumer, every ~30 seconds (by default), the consumer will check to see which consumers have a claim for each partition, then run some logic to determine whether it needs to 'steal' a partition from another consumer.  The service mechanism used to assert exclusive ownership over a partition is known as the [Epoch][Epoch].
+
+However, if no instances are being added or removed, there is an underlying issue that should be addressed.  See [Partition ownership changes a lot](#partition-ownership-changes-a-lot) for additional information and [Filing GitHub issues](#filing-github-issues).
+
+### High CPU usage
+
+High CPU usage is usually because an instance owns too many partitions.  We recommend no more than three partitions for every 1 CPU core; better to start with 1.5 partitions for each CPU core and test increasing the number of partitions owned.
+
+### Processor client stops receiving
+
+The processor client often is continually running in a host application for days on end.  Sometimes, they notice that EventProcessorClient is not processing one or more partitions.  Usually, this is not enough information to determine why the exception occurred.  The EventProcessorClient stopping is the symptom of an underlying cause (i.e. race condition) that occurred while trying to recover from a transient error.  Please see [Filing Github issues](#filing-github-issues) for the information we require.
+
+### Migrate from legacy to new client library
+
+The [migration guide][MigrationGuide] includes steps on migrating from the legacy client and migrating legacy checkpoints.
+
+## Get additional help
+
+Additional information on ways to reach out for support can be found in the [SUPPORT.md][SUPPORT] at the repo's root.
+
+### Filing GitHub issues
+
+When filing GitHub issues, the following details are requested:
+
+* Event Hub environment
+  * How many partitions?
+* EventProcessorClient environment
+  * What is the machine(s) specs processing your Event Hub?
+  * How many instances are running?
+  * What is the max heap set?
```

Comment:
don't know if the last two questions are relevant for us

Reply:
Repro steps and logs?   

I'm guessing that you mean from L195 and L196.   The instance count is needed, as that tells you how many partitions are owned.   The heap size is a Java detail that probably isn't relevant here.